sdks/python/apache_beam/io/fileio.py

```diff
@@ -139,12 +139,23 @@ def close(self, file_handle):
   @check_accessible(['file_path_prefix', 'file_name_suffix'])
   def initialize_write(self):
     file_path_prefix = self.file_path_prefix.get()
-    file_name_suffix = self.file_name_suffix.get()
-    tmp_dir = file_path_prefix + file_name_suffix + time.strftime(
-        '-temp-%Y-%m-%d_%H-%M-%S')
+
+    tmp_dir = self._create_temp_dir(file_path_prefix)
     FileSystems.mkdirs(tmp_dir)
     return tmp_dir
 
+  def _create_temp_dir(self, file_path_prefix):
+    base_path, last_component = FileSystems.split(file_path_prefix)
+    if not last_component:
+      raise ValueError('Cannot create a temporary directory for root path'
```

Comment:
Actually, there is a bug here. I should try to re-split the base path to make sure that this is a root. Lemme fix that.